service/http/handler.go

```diff
@@ -178,4 +178,13 @@ func (h *Handler) resolveIP(r *Request) {
 	if r.Header.Get("X-Real-Ip") != "" {
 		r.RemoteAddr = fetchIP(r.Header.Get("X-Real-Ip"))
 	}
+
+	if r.Header.Get("True-Client-IP") != "" {
+		r.RemoteAddr = fetchIP(r.Header.Get("True-Client-IP"))
+		return
+	}
+
+	if r.Header.Get("CF-Connecting-IP") != "" {
+		r.RemoteAddr = fetchIP(r.Header.Get("CF-Connecting-IP"))
+	}
```

Comment:
@vsychov I mean, in this case, if the client has `"X-Real-Ip"` you'll go through 3 checks `r.Header.Get("X-Real-Ip")` which will set the `r.RemoteAddr` -> `r.Header.Get("True-Client-IP")` and `r.Header.Get("CF-Connecting-IP")` which both just check for related headers w/o setting the remote address.
In the case of CF, the path would be the following: `r.Header.Get("X-Real-Ip")` -> `r.Header.Get("True-Client-IP")` -> return, 2 checks.
I propose to put the `return` in the `r.Header.Get("X-Real-Ip")` (hot path) condition, so, the path would be the following:
If the client has the  `r.Header.Get("X-Real-Ip")` -> return, only 1 check. In the case of CF: `r.Header.Get("X-Real-Ip")` -> `r.Header.Get("True-Client-IP")`, 2 checks. It would be 3:2 vs 1:2.
I agree about Enterprise, good catch, you can leave `return` stm in the  `True-Client-IP` condition, but also, could you please add another one in the `X-Real-Ip`?

Reply:
nevermind, I fixed this myself :)